CDCgov/data-exchange-portal
# Portal 1134/upload status endpoint

Adds a middleware API endpoint for fetching upload status for last 30 days from the supplemental API.
Endpoint reachable at /api/upload/status/{destinatinoId}.
Endpoint requires auth token.

Manual testing steps:
1. Make sure your local dev environment is set up following the README.md at the project root and in the /api directory
2. Run `yarn simulate` to start your local dev server and reverse proxy
3. Obtain an auth token by navigating to localhost:4280, logging in with sams, and inspecting your browser session storage
4. Send a request (postman or curl) to localhost:4280/api/upload/status/{destinationID}
- Obtain a valid destinationID from this file: https://github.com/CDCgov/data-exchange-upload/blob/main/tus/file-hooks/metadata-verify/allowed_destination_and_events.json
- Set the Authorization header to `Bearer {auth token}`, providing the auth token obtained from step 3

Automated testing steps:
1. CD into the /api directory
2. Follow directions in the README.md
3. run `npm run test`
